core/templates/components/forms/custom-forms-directives/image-uploader-modal.component.ts

```diff
@@ -0,0 +1,224 @@
+    private windowDimensionService: WindowDimensionsService,
+    private svgSanitizerService: SvgSanitizerService
+  ) {
+    super(ngbActiveModal);
+  }
+
+  private _getAspectRatio(): number {
+    return (
+      parseInt(this.aspectRatio.split(':')[0]) /
+      parseInt(this.aspectRatio.split(':')[1])
+    );
+  }
+
+  isThumbnail(): boolean {
+    return this.imageName === 'Thumbnail';
+  }
+
+  imageNotUploaded(): boolean {
+    return !this.uploadedImage && !this.thumbnailImageDataUrl;
+  }
+
+  initializeCropper(): void {
+    if (!this.croppableImageRef) {
+      return;
+    }
+    let imageElement = this.croppableImageRef.nativeElement;
+    if (this.windowIsNarrow) {
+      this.cropper = new Cropper(imageElement, {
+        minContainerWidth: 200,
+        minContainerHeight: 200,
+        aspectRatio: this._getAspectRatio(),
+      });
+    } else {
+      this.cropper = new Cropper(imageElement, {
+        minContainerWidth: 500,
+        minContainerHeight: 350,
+        aspectRatio: this._getAspectRatio(),
+      });
+    }
+  }
+
+  onFileChanged(file: Blob): void {
+    this.invalidImageWarningIsShown = false;
+    let reader = new FileReader();
+    reader.onload = e => {
+      this.invalidTagsAndAttributes = {
+        tags: [],
+        attrs: [],
+      };
+      let imageData = (e.target as FileReader).result as string;
+      if (this.svgSanitizerService.isBase64Svg(imageData)) {
+        this.invalidTagsAndAttributes =
+          this.svgSanitizerService.getInvalidSvgTagsAndAttrsFromDataUri(
+            imageData
+          );
+        this.thumbnailImageDataUrl =
+          this.svgSanitizerService.removeAllInvalidTagsAndAttributes(imageData);
+        this.uploadedImage =
+          this.svgSanitizerService.getTrustedSvgResourceUrl(imageData);
+      }
+      if (!this.uploadedImage) {
+        this.uploadedImage = decodeURIComponent(
+          (e.target as FileReader).result as string
+        );
+      }
+
+      const img = new Image();
+      img.onload = () => {
+        this.dimensions = {height: img.height, width: img.width};
+
+        try {
+          this.changeDetectorRef.detectChanges();
+        } catch (viewDestroyedError) {
+          // This try catch block handles the following error in FE tests:
+          // ViewDestroyedError:
+          //   Attempt to use a destroyed view: detectChanges thrown.
+          // No further action is needed.
+        }
+        if (this.imageName !== 'Thumbnail') {
+          this.initializeCropper();
+        }
+      };
+      img.src = imageData;
+      this.imageType = file.type;
+    };
+    reader.readAsDataURL(file);
+  }
+
+  reset(): void {
+    this.invalidTagsAndAttributes = {
+      tags: [],
+      attrs: [],
+    };
+    this.uploadedImage = null;
+    this.croppedImageDataUrl = null;
+    this.thumbnailImageDataUrl = null;
+  }
+
+  onInvalidImageLoaded(): void {
+    this.reset();
+    this.invalidImageWarningIsShown = true;
+  }
```

Comment:
This function's never called?

Reply:
Thanks yes this should be called if we find any error reading the file. Done added it in reader.onerror